command.go

```diff
@@ -104,46 +147,72 @@ func (c *Command) Run(ctx *Context) (err error) {
 		c.UseShortOptionHandling = true
 	}
 
-	set, err := c.parseFlags(ctx.Args(), ctx.shellComplete)
+	c.categories = newCommandCategories()
+	for _, command := range c.Subcommands {
+		c.categories.AddCommand(command.Category, command)
+	}
+	sort.Sort(c.categories.(*commandCategories))
+
+	var newCmds []*Command
+	for _, scmd := range c.Subcommands {
+		if scmd.HelpName == "" {
+			scmd.HelpName = fmt.Sprintf("%s %s", c.HelpName, scmd.Name)
+		}
+		newCmds = append(newCmds, scmd)
+	}
+	c.Subcommands = newCmds
+}
+
+func (c *Command) Run(cCtx *Context, arguments []string) (err error) {
```

Comment:
This method signature is an API change. Could it instead use variadic for `arguments` so that existing code can still compile?

```go
func (c *Command) Run(cCtx *Context, arguments ...string) (err error) {
```

Reply:
Makes sense. Updated.